torch/csrc/jit/tensorexpr/llvm_codegen.cpp

```diff
@@ -77,7 +77,7 @@ llvm::CmpInst::Predicate llvm_comparison_predicate(
 
 class LLVMCodeGenImpl : public IRVisitor {
  private:
-  llvm::orc::ThreadSafeContext context_;
+  std::unique_ptr<llvm::LLVMContext> context_;
```

Comment:
a random question; could we move it to `PytorchLLVMJIT`? We seem to be passing `context_` into `jit_` in a few function calls below.

Reply:
We use the context_ a lot in LLVMCodeGen for things like creating basic blocks and types.  Something that's awkward in this code is that we keep context_ and module_ as members of LLVMCodeGen (since they're useful to have in building the IR, obvs) but then we move them into LLVMJIT and they're left as kind-of-dangling pointers in the LLVMCodeGen object.  They're never used again (the only thing you do with a CodeGen after constructing it is `call` it, which doesn't do anymore codegen) but it's still sloppy engineering.

Reply:
Fixing it is kinda beyond the scope of what I'm trying to do here though.